lib/service/src/main/java/io/atlasmap/service/MappingService.java

```diff
@@ -91,6 +97,126 @@ public MappingService(AtlasService parent) {
         this.previewContext = this.atlasService.getContextFactory().createPreviewContext();
     }
 
+    /**
+     * Return the field from the specified list of mapped fields whose index property matches the
+     * specified field index.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return field
+     */
+    private Field getFieldByIndex(List<Field> mappedFields, int fieldIndex) {
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                return field;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return the ordinal position of the mapped field with the specified index property.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return ordinal position
+     */
+    private int getOrdinalPosition(List<Field> mappedFields, int fieldIndex) {
+        int ordinalPosition = 0;
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                break;
+            }
+            ordinalPosition++;
+        }
+        return ordinalPosition;
+    }
+
+    /**
+     * Return the mapping element for the specified mapping uuid.
+     *
+     * @param uuid
+     * @returns mapping element
+     */
+    private Mapping getMappingByID(List<BaseMapping> mappings, String uuid) {
+        Mapping objectMapping = null;
+        if (mappings != null) {
+            for (ListIterator<BaseMapping> iter = mappings.listIterator(); iter.hasNext();) {
+                Mapping element = (Mapping)iter.next();
+                if (element.getId().equals(uuid)) {
+                    objectMapping = element;
+                    break;
+                }
+            }
+        }
+        return objectMapping;
+    }
+
+    /**
+     * Change the value of the specified current index to the specified target index for the specified
+     * mapped field (source/target).
+     *
+     * @param mappingDefinitionId
+     * @param mappingId
+     * @param dataSourceType
+     * @param currentFieldIndex
+     * @param targetFieldIndex
+     *
+     * @return empty response
+     */
+    @PUT
+    @Path("/{mappingId}/field/{dataSourceType}/current/{currentFieldIndex}/targetFieldIndex")
```

Comment:
Better to be `/{mappingId}/field/{dataSourceType}/{fieldIndex}/index`
URL represents the location of the resource, therefore it shouldn't contain something operation specific. The `current` in the path sounds like there is such property on a mapping field.
For example, if we were to add an operation to get/set the `FieldType` on a mapping field, the URL should be
`/{mappingId}/field/{dataSourceType}/{fieldIndex}/fieldType`
where `GET` method gets the current value and `PUT` updates the `FieldType` while URL is same. Note that only the last part of the path is different from index one. We won't have `GET` for the field index as it's self-evident, but the URL rule stays same.

Reply:
What about `targetFieldIndex` - even though that exists in the payload - doesn't it still need to be specified in the URL?

Reply:
`/{mappingId}/field/{dataSourceType}/{fieldIndex}/index`
so the `targetFieldIndex` should be instead `index` as a property of the mapping field.